xmtp_api/src/identity.rs

```diff
@@ -93,30 +98,44 @@ where
         Ok(res)
     }
 
-    #[tracing::instrument(level = "debug", skip(self), fields(len = account_addresses.len()))]
+    #[tracing::instrument(level = "debug", skip(self), fields(len = account_identifiers.len()))]
     pub async fn get_inbox_ids(
         &self,
-        account_addresses: Vec<String>,
-    ) -> Result<AddressToInboxIdMap> {
+        account_identifiers: Vec<ApiIdentifier>,
+    ) -> Result<IdentifierToInboxIdMap> {
         tracing::info!(
-            "Getting inbox_ids for account addresses: {:?}",
-            &account_addresses
+            "Getting inbox_ids for account identities: {:?}",
+            &account_identifiers
         );
+        let requests = account_identifiers
+            .into_iter()
+            .map(|r| GetInboxIdsRequestProto {
+                identifier: r.identifier,
+                identifier_kind: r.identifier_kind as i32,
+            })
+            .collect();
         let result = self
             .api_client
-            .get_inbox_ids(GetInboxIdsRequest {
-                requests: account_addresses
-                    .into_iter()
-                    .map(|address| GetInboxIdsRequestProto { address })
-                    .collect(),
-            })
+            .get_inbox_ids(GetInboxIdsRequest { requests })
             .await
             .map_err(ApiError::from)?;
 
         Ok(result
             .responses
             .into_iter()
-            .filter_map(|resp| Some((resp.address, resp.inbox_id?)))
+            .filter_map(|resp| {
+                let kind = match resp.identifier_kind() {
+                    IdentifierKind::Unspecified => IdentifierKind::Ethereum,
```

Comment:
Q: do we want to double check if it's really Eth?

Reply:
Since this is a new field, if it comes in as unspecified, it means it's from an old version of the proto definitions. And since LibXMTP before now is only Eth, I think it's safe to assume that it's Eth :) After this PR, it should always be specified.